scipy/interpolate/_ppoly.pyx

```diff
@@ -914,6 +922,9 @@ cdef int croots_poly1(double[:,:,::1] c, double y, int ci, int cj,
         nworkspace = n*n + lwork
         workspace[0] = libc.stdlib.malloc(nworkspace * sizeof(double))
 
+        if workspace[0] == NULL:
+            raise MemoryError("Failed to allocate memory in croots_poly1")
```

Comment:
This is in a `cdef` function with return type `int`, so using `raise` like this won't work the way we want it to.  If this statement is executed,the exception is not raised.  Instead, a warning is printed and the function returns 0.  You can read more about this at http://docs.cython.org/en/latest/src/userguide/language_basics.html#error-return-values and http://docs.cython.org/en/latest/src/tutorial/clibraries.html#handling-errors